lib/login.js

```diff
@@ -90,8 +91,9 @@ module.exports = {
 
         // If no entry for the auth token is found, add one
         if (authWrite === -1) {
-            lines.push(args.registry.slice(args.registry.search(/\:\/\//, '') +
-            1) + '/:_authToken=' + (args.quotes ? '"' : '') + response.token + (args.quotes ? '"' : ''));
+            const obtainedRegistry = args.registry.slice(args.registry.search(/\:\/\//, '') + 1)
+            const updatedRegistry = obtainedRegistry.endsWith('/') ? obtainedRegistry.slice(0, -1) : obtainedRegistry;
```

Comment:
Let's extract the trailing-slash-stripping out so that both blocks can share it, and then also we can do it all with one variable. What do you think?

```diff
diff --git a/lib/login.js b/lib/login.js
index 38f4ba0..8bd6585 100644
--- a/lib/login.js
+++ b/lib/login.js
@@ -78,9 +78,15 @@ module.exports = {
             }
         }
 
+        let registry = args.registry.slice(args.registry.search(/\:\/\//) + 1);
+
+        // Trim trailing slash before a new one is added later.
+        if (registry.endsWith('/')) {
+            registry = registry.slice(0, -1);
+        }
+
         var authWrite = lines.findIndex(function (element, index, array) {
-            if ( element.indexOf(args.registry.slice(args.registry.search(/\:\/\//, '') + 1) +
-            '/:_authToken=') !== -1) {
+            if (element.includes(`${registry}/:_authToken=`)) {
                 // If an entry for the auth token is found, replace it
                 array[index] = element.replace(/authToken\=.*/, 'authToken=' + (args.quotes ? '"' : '') +
                 response.token + (args.quotes ? '"' : ''));
@@ -90,8 +96,7 @@ module.exports = {
 
         // If no entry for the auth token is found, add one
         if (authWrite === -1) {
-            lines.push(args.registry.slice(args.registry.search(/\:\/\//, '') +
-            1) + '/:_authToken=' + (args.quotes ? '"' : '') + response.token + (args.quotes ? '"' : ''));
+            lines.push(registry + '/:_authToken=' + (args.quotes ? '"' : '') + response.token + (args.quotes ? '"' : ''));
         }
 
         var toWrite = lines.filter(function (element) {
```